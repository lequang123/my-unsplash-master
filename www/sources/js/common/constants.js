import LanguageHelper from './langHelper';

export const PAGING_SIZES = [100, 200, 300, 400, 500, 1000];

export const DefaultPageSize = 100;

export const DateFormat = 'MM/dd/yyyy';

export const DateTimeFormat = 'MM/dd/yyyy hh:mm:ss a';

export const TimeFormat = 'hh:mm:ss a';

export const MappingStatus = {
    'BotRemoved': 0,
    'NotMapped': 1,
    'Mapped': 2
};

export const ChatTypeMapping = {
    'Personal': 2,
    'Group': 1
};

export const MULTI_SELECT_TEXT_OPTIONS = {
    Texts: LanguageHelper.getResource('All'),
    SelectedItemCount: '# selected',
    SelectOptions: LanguageHelper.getResource('PleaseSelect'),
    NoData: LanguageHelper.getResource('NoData')
};

export const AUTO_REFRESH_TIMES = [90, 120, 150, 180];