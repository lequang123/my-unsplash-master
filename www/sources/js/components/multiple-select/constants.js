export const KEY_NAME = 'key';
export const VALUE_NAME = 'value';
export const STATUS_NAME = 'checked';
export const MAX_SELECTED_ITEM_FOR_DISPLAY = 3;
export const DEFAULT_CHECKED = false;