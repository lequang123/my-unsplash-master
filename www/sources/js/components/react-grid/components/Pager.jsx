import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Pager extends Component {
    static propTypes = {
        pageOption: PropTypes.shape({
            PageIndex: PropTypes.number,
            PageSize: PropTypes.number,
            PageList: PropTypes.arrayOf(PropTypes.number),
            TotalItem: PropTypes.number
        }),
        onPaging: PropTypes.func,
        isShowPagingInfo: PropTypes.bool,
        isAllowInputPageIndex: PropTypes.bool
    };

    static defaultProps = {
        pageOption: {
            PageIndex: 1,
            PageSize: 50,
            PageList: [50, 100, 200],
            TotalItem: 0
        },
        isShowPagingInfo: true,
        isAllowInputPageIndex: true
    };

    constructor(props) {
        super(props);

        this.state = {
            inputValue: props.pageOption.PageIndex
        };
    }

    componentDidUpdate(prevProps) {
        const pageIndex = this.props.pageOption.PageIndex;
        if (pageIndex !== prevProps.pageOption.PageIndex) {
            this._updateState(pageIndex);
        }
    }

    goToPage(index, pageSize) {
        this.props.onPaging && this.props.onPaging(index, pageSize);
    }

    onKeyUpHandle = (e, totalPage) => {
        const code = e.keyCode ? e.keyCode : e.which;
        if (code === 13) {
            const value = parseInt(this.state.inputValue, 10);
            if (value > 0 && value <= totalPage) {
                this.goToPage(value, this.props.pageOption.PageSize);
            }
        }
    };

    onClickHandle(isDisable, handleFunc) {
        if (!isDisable) {
            handleFunc();
        }
    }

    onInputChangeHandler = event => {
        const pattern = /^\d*$/;
        const regex = new RegExp(pattern);
        const { value } = event.target;
        const isValidNumber = regex.test(value); // NOSONAR
        if (isValidNumber) {
            this._updateState(value);
        }
    };

    _updateState(inputValue) {
        this.setState({ inputValue });
    }

    _renderPageSizeSelection(isDisable, pageSize) {
        return (
            <select className="pagesize-select" disabled={isDisable} value={pageSize}
                onChange={event => this.goToPage(1, Number(event.target.value))}
            >
                {this._renderPageSizeOption(this.props.pageOption.PageList)}
            </select>
        );
    }

    _renderPageSizeOption(pageList) {
        return pageList.map((item, index) => <option key={`pagesize${index}`} value={item}>{item}</option>);
    }

    _renderPageIcon(iconClass, isDisable, onClickFunc) {
        return (
            <a className={`fa ${iconClass} ${isDisable ? 'disable' : ''}`}
                onClick={() => this.onClickHandle(isDisable, onClickFunc)}
            />
        );
    }

    _renderPageInput(index, totalPage) {
        return (
            <div className="page-input">
                <span>Page </span>
                <input type="text" className="page-form" value={index} maxLength="4" size="2"
                    onChange={this.onInputChangeHandler}
                    onKeyUp={event => this.onKeyUpHandle(event, totalPage)}
                />
                <span>{` of ${totalPage}`} </span>
            </div>
        );
    }

    _renderPageLabel(index, totalPage) {
        return (<div className="page-input"><span>Page {`${index} of ${totalPage} `}</span></div>);
    }

    _renderPageInfo(index, pageSize, totalItem) {
        const viewFrom = ((index - 1) * pageSize) + 1;
        const viewTo = index * pageSize;
        return (<span className="page-info">{`Display ${viewFrom} to ${viewTo > totalItem ? totalItem : viewTo} of ${totalItem} item(s)`}</span>);
    }

    render() {
        const { isShowPagingInfo, isAllowInputPageIndex, pageOption, minWidth, maxWidth } = this.props;
        const { PageIndex, PageSize, TotalItem } = pageOption;
        const { inputValue } = this.state;
        const isShow = TotalItem > 0;
        const isDisable = isNaN(parseInt(inputValue, 10));
        const totalPage = Math.ceil(TotalItem / PageSize);

        return (
            <div className="page-container" style={{ minWidth, maxWidth, display: `${isShow ? 'block' : 'none'}` }}>
                Page size {this._renderPageSizeSelection(isDisable, PageSize)}
                {this._renderPageIcon('fa-angle-double-left', PageIndex === 1, () => this.goToPage(1, PageSize))}
                {this._renderPageIcon('fa-angle-left', isDisable || PageIndex <= 1 || PageIndex > totalPage, () => this.goToPage(PageIndex - 1, PageSize))}
                {isAllowInputPageIndex ? this._renderPageInput(inputValue, totalPage) : this._renderPageLabel(PageIndex, totalPage)}
                {this._renderPageIcon('fa-angle-right', isDisable || PageIndex >= totalPage || PageIndex < 1, () => this.goToPage(PageIndex + 1, PageSize))}
                {this._renderPageIcon('fa-angle-double-right', PageIndex === totalPage, () => this.goToPage(totalPage, PageSize))}
                {isShowPagingInfo && this._renderPageInfo(PageIndex, PageSize, TotalItem)}
            </div>
        );
    }
}