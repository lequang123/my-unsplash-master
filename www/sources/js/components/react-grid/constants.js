export const SCROLLBAR_WIDTH = 19;
export const GUTTER = 15;
export const DEFAULT_COLUMN_WIDTH = 100;