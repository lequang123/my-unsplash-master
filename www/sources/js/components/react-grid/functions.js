export const scrollToTop = (element, scrollDuration) => {
    const scrollStep = -element.scrollTop / (scrollDuration / 15);
    requestAnimationFrame(() => {
        let scrollInterval = null;
        scrollInterval = setInterval(() => {
            if (element.scrollTop !== 0) {
                element.scrollTop += scrollStep;
            } else {
                clearInterval(scrollInterval);
            }
        }, 15);
    });
};

const functions = {
    scrollToTop
};

export default functions;