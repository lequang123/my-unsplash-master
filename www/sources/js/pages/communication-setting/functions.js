export const getSubCategories = (isCheckDefault, subCategoryList, selectedSubCategoryIds) => {
    const selectedIdsDistinct = [];
    const distinctSub = [];
    subCategoryList.map(sub => {
        if (selectedIdsDistinct.indexOf(sub.subCategoryId) < 0) {
            selectedIdsDistinct.push(sub.subCategoryId);
            distinctSub.push(sub);
        }
    });

    const distinctSubCategories = distinctSub.map(sub => {
        return {
            name: sub.name,
            subCategoryId: sub.subCategoryId,
            checked: isCheckDefault ? true : selectedSubCategoryIds.indexOf(sub.subCategoryId.toString()) >= 0 ? true : false
        };
    });

    return distinctSubCategories;
};

const functions = {
    getSubCategories
};