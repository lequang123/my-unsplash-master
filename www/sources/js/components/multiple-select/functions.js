import { KEY_NAME, VALUE_NAME, STATUS_NAME } from './constants';

export const convertDataSourceToState = ({ keyField, valueField, statusField, dataSource }) => {
    return dataSource.map(item => {
        return {
            [KEY_NAME]: item[keyField],
            [VALUE_NAME]: item[valueField],
            [STATUS_NAME]: item[statusField] ? item[statusField] : false,
            subData: item.subData ? item.subData : [],
            visible: true
        };
    });
};

const functions = {
    convertDataSourceToState
}

export default functions;