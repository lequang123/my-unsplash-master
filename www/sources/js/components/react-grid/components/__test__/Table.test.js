import React from 'react';
import Table from '../Table';
import RowLayout from '../RowLayout';
import createTableSection from '../tableSection';
import { DEFAULT_COLUMN_WIDTH, SCROLLBAR_WIDTH } from '../..//constants';
import Pager from '../Pager';
import functions from '../../functions';
import Site from '@jsroot/common/site';
import { shallow } from 'enzyme';
import { Tab } from 'bootstrap';

describe('Table', () => {
    const clientWidth = document.body.clientWidth;

    it('should have correct snapshot', () => {
        // Arrange
        const props = {
            header: {
                props: {
                    children: []
                }
            }
        };

        // Action
        const table = shallow(<Table {...props} />);

        // Assert
        expect(table).toMatchSnapshot();
    });

    it('componentDidMount should run correctly', () => {
        // Arrange
        const props = {
            header: {
                props: {
                    children: []
                }
            }
        };

        const debounceResizing = 'debounceResizing';
        const table = shallow(<Table {...props} />);
        table.instance()._handleResize = jest.fn();
        table.instance().debounceResizing = debounceResizing;
        window.addEventListener = jest.fn();

        // Action
        table.instance().componentDidMount();

        // Assert
        expect(table.instance()._handleResize).toHaveBeenCalled();
        expect(window.addEventListener).toHaveBeenCalledWith('resize', debounceResizing);
    });

    describe('componentDidUpdate', () => {
        it('should change state when maxWidth change', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                },
                maxWidth: 5
            };
    
            const prevProps = { maxWidth: 10 };
            const maxWidthValue = props.maxWidth || clientWidth;
            const diffWidth = clientWidth - maxWidthValue;
            const table = shallow(<Table {...props} />);

            // Action
            table.instance().componentDidUpdate(prevProps);

            // Assert
            expect(table.instance().state.maxWidth).toEqual(maxWidthValue);
            expect(table.instance().diffWidth).toEqual(diffWidth);
        });

        it('should change state when maxWidth change and header change', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                },
                maxWidth: 5
            };
    
            const prevProps = {
                header: {
                    props: {
                        children: [{}]
                    }
                },
                maxWidth: 10
            };

            const columnsWidth = 15;
            const table = shallow(<Table {...props} />);
            table.instance()._getColumnsWidth = jest.fn(() => columnsWidth);

            // Action
            table.instance().componentDidUpdate(prevProps);

            // Assert
            expect(table.instance().state.columnsWidth).toEqual(columnsWidth);
        });

        it('should scrollToTop when body change and shouldResetScrollPosition true', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                },
                shouldResetScrollPosition: true
            };
    
            const prevProps = {
                body: []
            };

            const table = shallow(<Table {...props} />);
            const bodyWrapper = jest.fn();
            table.instance().bodyWrapper = bodyWrapper;
            functions.scrollToTop = jest.fn();

            // Action
            table.instance().componentDidUpdate(prevProps);

            // Assert
            expect(functions.scrollToTop).toHaveBeenCalledWith(bodyWrapper, 200);
        });
    });

    it('componentWillUnmount should run correctly', () => {
        // Arrange
        const props = {
            header: {
                props: {
                    children: []
                }
            }
        };

        const table = shallow(<Table {...props} />);
        const debounceResizing = 'debounceResizing';
        table.instance().debounceResizing = debounceResizing;
        window.removeEventListener = jest.fn();

        // Action
        table.instance().componentWillUnmount();

        // Assert
        expect(window.removeEventListener).toHaveBeenCalledWith('resize', debounceResizing);
    });

    describe('columnWidthSum', () => {
        it('should return null when columnsWidth false', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                }
            };
    
            const state = { columnsWidth: false };
            const table = shallow(<Table {...props} />);
            table.setState(state);

            // Action
            const result = table.instance().columnWidthSum;

            // Assert
            expect(result).toBeNull;
        });
        
        it('should return null when columnsWidth element type not number', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                }
            };
    
            const state = { columnsWidth: [ 'value' ] };
            const table = shallow(<Table {...props} />);
            table.setState(state);

            // Action
            const result = table.instance().columnWidthSum;

            // Assert
            expect(result).toBeNull;
        });
        
        it('should return expected when columnsWidth element type always number', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                }
            };
    
            const state = { columnsWidth: [ 1, 2 ] };
            const table = shallow(<Table {...props} />);
            table.setState(state);

            // Action
            const result = table.instance().columnWidthSum;

            // Assert
            expect(result).toEqual(3);
        });
    });

    it('_getNextColspanCells should return expected', () => {
        // Arrange
        const props = {
            header: {
                props: {
                    children: []
                }
            }
        };

        const cells = [
            {
                props: {
                    colSpan: 1
                }
            },
            {
                props: {
                    colSpan: 2
                }
            }
        ];

        const fromCellIndex = 0;
        const toCellIndex = 3;
        const table = shallow(<Table {...props} />);

        // Action
        const result = table.instance()._getNextColspanCells(cells, fromCellIndex, toCellIndex);

        // Assert
        expect(result).toEqual(cells);
    });

    describe('_getColumnsWidth', () => {
        it('should return null when empty array headerRows', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                }
            };

            const headerRows = [];
            const table = shallow(<Table {...props} />);

            // Action
            const result = table.instance()._getColumnsWidth(headerRows);

            // Assert
            expect(result).toBeNull;
        });

        it('should return columnsWidth when not empty array headerRows', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                }
            };

            const cellList = [];
            const headerRows = [
                {
                    props: {
                        children: cellList
                    }
                }
            ];

            let columnsWidth = [];
            const table = shallow(<Table {...props} />);
            table.instance()._getWidthByCells = jest.fn();

            // Action
            table.instance()._getColumnsWidth(headerRows);

            // Assert
            expect(table.instance()._getWidthByCells).toHaveBeenCalledWith(headerRows, columnsWidth, {}, cellList, 0);
        });

        it('should return columnsWidth when not array headerRows', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                }
            };

            const cellList = [];
            const headerRows = {
                props: {
                    children: cellList
                }
            };

            let columnsWidth = [];
            const table = shallow(<Table {...props} />);
            table.instance()._getWidthByCells = jest.fn();

            // Action
            table.instance()._getColumnsWidth(headerRows);

            // Assert
            expect(table.instance()._getWidthByCells).toHaveBeenCalledWith(headerRows, columnsWidth, {}, cellList, 0);
        });
    });

    it('_getWidthByCells should run correctly', () => {
        // Arrange
        const props = {
            header: {
                props: {
                    children: []
                }
            }
        };

        const cell = {};
        const cellList = [cell, null];
        const headerRows = {
            props: {
                children: cellList
            }
        };

        const columnsWidth = [];
        const currentRowIndex = 0;
        const currentNextCellIndexByRow = {};
        const table = shallow(<Table {...props} />);
        table.instance()._handleWidthOneCell = jest.fn();

        // Action
        table.instance()._getWidthByCells(headerRows, columnsWidth, currentNextCellIndexByRow, cellList, currentRowIndex);

        // Assert
        expect(table.instance()._handleWidthOneCell).toHaveBeenCalledTimes(1);
        expect(table.instance()._handleWidthOneCell).toHaveBeenCalledWith(headerRows, columnsWidth, currentNextCellIndexByRow, cell, currentRowIndex);
    });

    describe('_handleWidthOneCell', () => {
        it('should run correctly when colspan greater than 1', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                }
            };
    
            const cell = {
                props: {
                    colSpan: 2
                }
            };

            const headerRows = [
                {
                    props: {
                        children: []
                    }
                },
                {
                    props: {
                        children: []
                    }
                }
            ];

            const columnsWidth = [];
            const currentRowIndex = 0;
            const currentNextCellIndexByRow = {};
            const table = shallow(<Table {...props} />);
            table.instance()._getWidthByCells = jest.fn();
            table.instance()._getNextColspanCells = jest.fn(() => []);

            // Action
            table.instance()._handleWidthOneCell(headerRows, columnsWidth, currentNextCellIndexByRow, cell, currentRowIndex);

            // Assert
            expect(table.instance()._getWidthByCells).toHaveBeenCalledWith(headerRows, columnsWidth, currentNextCellIndexByRow, [], currentRowIndex + 1);
        });
    });

    describe('_handleResize', () => {
        it('should change state contentHeight', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                },
                autoHeight: true
            };

            const state = { contentHeight: 10 };
            const event = { preventDefault: jest.fn() };
            const table = shallow(<Table {...props} />);
            const expected = 20;
            table.instance().bodyWrapper = true;
            table.instance().setState(state);
            table.instance()._calculateBodyHeight = jest.fn(() => expected);

            // Action
            table.instance()._handleResize(event);

            // Assert
            expect(event.preventDefault).toHaveBeenCalled();
            expect(table.instance().state.contentHeight).toEqual(expected);
        });
        
        it('should change state maxWidth', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                }
            };

            const state = { maxWidth: 123 };
            const diffWidth = 10;
            const maxWidth = document.body.clientWidth - diffWidth
            const event = { preventDefault: jest.fn() };
            const table = shallow(<Table {...props} />);
            table.instance().diffWidth = diffWidth;
            table.instance().setState(state);

            // Action
            table.instance()._handleResize(event);

            // Assert
            expect(table.instance().state.maxWidth).toEqual(maxWidth);
        });
    });

    it('_calculateBodyHeight should return expected', () => {
        // Arrange
        const props = {
            header: {
                props: {
                    children: []
                }
            }
        };

        const footerWrapper = { offsetHeight: 10 };
        const bodyWrapper = { offsetTop: 10 };
        const adjustedHeight = 10;
        const expected = window.innerHeight - (bodyWrapper.offsetTop + footerWrapper.offsetHeight + 70) - adjustedHeight;
        const table = shallow(<Table {...props} />);
        table.instance().footerWrapper = footerWrapper;
        table.instance().bodyWrapper = bodyWrapper;
        table.instance().adjustedHeight = adjustedHeight;

        // Action
        const result = table.instance()._calculateBodyHeight();

        // Assert
        expect(result).toEqual(expected);
    });

    it('_getUpdatedColumnLayout should return expected', () => {
        // Arrange
        const props = {
            header: {
                props: {
                    children: []
                }
            },
            width: 10,
            autoWidth: false,
            containerPadding: 10
        };

        const columnWidth = 10;
        const state = { columnsWidth: [ columnWidth ] };

        const sumOfColumnWidth = props.width - props.containerPadding - SCROLLBAR_WIDTH;
        const expected = state.columnsWidth.map(cellWidth =>
            sumOfColumnWidth / columnWidth * cellWidth);
        const table = shallow(<Table {...props} />);
        table.instance().setState(state);

        // Action
        const result = table.instance()._getUpdatedColumnLayout();

        // Assert
        expect(result).toEqual(expected);
    });

    describe('render', () => {
        it('should return null when columnsWidth empty', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                }
            };

            const state = { columnsWidth: [] };
            const table = shallow(<Table {...props} />);
            table.instance().setState(state);

            // Action
            const result = table.instance().render();

            // Assert
            expect(result).toBeNull;
        });
        
        it('should return expected components', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                },
                containerPadding: 10
            };

            const state = { maxWidth: 100 };
            const table = shallow(<Table {...props} />);
            table.instance().setState(state);

            // Action
            const result = shallow(table.instance().render());

            // Assert
            expect(result.find('div').at(1).props().className).toBe('table-container');
            expect(result.find('div').at(1).props().style).toEqual({ maxWidth: state.maxWidth - props.containerPadding });
        });
        
        it('should return expected components have Pager', () => {
            // Arrange
            const props = {
                header: {
                    props: {
                        children: []
                    }
                },
                isPaging: true,
                pageOption: {
                    PageIndex: 1
                },
                onPaging: jest.fn()
            };

            // Action
            const table = shallow(<Table {...props} />);

            // Assert
            expect(table.find(Pager).props().pageOption).toEqual(props.pageOption);
            expect(table.find(Pager).props().onPaging).toEqual(props.onPaging);
        });
    });
});