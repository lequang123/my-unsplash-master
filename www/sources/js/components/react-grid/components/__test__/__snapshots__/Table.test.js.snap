// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should have correct snapshot 1`] = `ShallowWrapper {}`;
